src/main/resources/net/uaznia/lukanus/hudson/plugins/gitparameter/javascript/git-parameter-select.js

```diff
@@ -4,74 +4,99 @@ function gitParameterUpdateSelect(listBox, url, divId, config) {
     config = config || {};
     config = object(config);
     var originalOnSuccess = config.onSuccess;
-    var l = $(listBox);
+    // TODO Inline this variable, maybe?
+    var l = listBox;
     // TODO - Remove '.firstChild?.nextSibling' once this plugin targets https://github.com/jenkinsci/jenkins/pull/6460 or higher
     var status = findFollowingTR(listBox, "validation-error-area").firstChild?.nextSibling || findFollowingTR(listBox, "validation-error-area");
     if (status.firstChild && status.firstChild.getAttribute('data-select-ajax-error')) {
         status.innerHTML = "";
     }
     config.onSuccess = function (rsp) {
-        l.removeClassName("select-ajax-pending");
+        l.classList.remove("select-ajax-pending");
         var currentSelection = l.value;
 
         // clear the contents
         while (l.length > 0) l.options[0] = null;
 
         var selectionSet = false; // is the selection forced by the server?
         var possibleIndex = null; // if there's a new option that matches the current value, remember its index
-        var opts = eval('(' + rsp.responseText + ')').values;
-        for (var i = 0; i < opts.length; i++) {
-            l.options[i] = new Option(opts[i].name, opts[i].value);
-            if (opts[i].selected) {
-                l.selectedIndex = i;
-                selectionSet = true;
+        rsp.json().then(json => {
```

Comment:
This exposes a pre-existing race condition that was not exposed under Prototype, as we discovered with the copypasta version of this code in Jenkins core in jenkinsci/jenkins#7951 and the subsequent revert (jenkinsci/jenkins#7965). The trick, as we discovered in jenkinsci/jenkins#7982, is to close the race condition window by moving `rsp.json().then(json => {` _above_ `l.classList.remove("select-ajax-pending");`.